pages/app/models/refinery/page.rb

```diff
@@ -378,11 +381,15 @@ def puts_destroy_help
     def slug_locale
       return Globalize.locale if translation_for(Globalize.locale, false).try(:slug).present?
 
-      if translations.empty? || translation_for(Refinery::I18n.default_frontend_locale, false).present?
+      if translations.empty? || translation_for(Refinery::I18n.default_frontend_locale, false).try(:slug).present?
         Refinery::I18n.default_frontend_locale
       else
         translations.first.locale
       end
     end
+
+    def update_all_descendants
+      self.descendants.update_all(updated_at: DateTime.now)
```

Comment:
I think you can achieve this with the `touch` method http://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-touch which will set the `updated_at` for you without any timezone worries

Reply:
I completly forget this method, that's a lot cleaner!